packages/hash/frontend/src/components/BlockLoader/data-map-editor.tsx

```diff
@@ -306,26 +313,22 @@ export const DataMapEditor = ({
                   })}
                 </Select>
               </Box>
-              <Box width="30%" ml={4}>
+              <Box flex={1} ml={4}>
                 <TextField
                   size="small"
-                  value={schemaMap?.transformations?.[targetPath]?.default}
+                  value={schemaMap.transformations?.[targetPath]?.default}
                   onChange={(evt) =>
-                    updateSchemaMap((existingMap) => ({
-                      ...(existingMap ?? { mapId }),
-                      transformations: {
-                        ...existingMap.transformations,
-                        [targetPath]:
-                          evt.target.value != null
-                            ? {
-                                default: evt.target.value,
-                                sourceKey:
-                                  existingMap.transformations?.[targetPath]
-                                    ?.sourceKey,
-                              }
-                            : undefined,
-                      },
-                    }))
+                    onSchemaMapChange((existingMap) =>
+                      produce(existingMap, (draftMap) => {
+                        if (evt.target.value == null) {
+                          return;
+                        }
+                        draftMap.transformations ??= {};
+                        draftMap.transformations[targetPath] ??= {};
+                        draftMap.transformations[targetPath]!.default =
+                          evt.target.value;
+                      }),
+                    )
```

Comment:
Immer has a great feature where when you only pass it a function, it returns a function, which means you can do this

```suggestion
                    onSchemaMapChange(produce((draftMap) => {
                      if (evt.target.value == null) {
                        return;
                      }
                      draftMap.transformations ??= {};
                      draftMap.transformations[targetPath] ??= {};
                      draftMap.transformations[targetPath]!.default =
                        evt.target.value;
                    }))
```